File: BE09-Final-2team-BE/product-service/src/main/java/com/momnect/productservice/command/entity/product/SortOption.java
package com.momnect.productservice.command.entity.product;

import lombok.Getter;

@Getter
public enum SortOption {
    RECOMMENDED("추천순"),
    LATEST("최신순"),
    PRICE_ASC("낮은가격순"),
    PRICE_DESC("높은가격순");

    private final String label;

    SortOption(String label) {
        this.label = label;
    }

    public static String getLabel(SortOption option) {
        return option.getLabel();
    }
}

File: BE09-Final-2team-FE/src/enums/sortOption.js
// 요청용 (백엔드 전달 값)
export const SortOption = Object.freeze({
    RECOMMENDED: 'RECOMMENDED', // 추천순
    LATEST: 'LATEST', // 최신순
    PRICE_ASC: 'PRICE_ASC', // 낮은가격순
    PRICE_DESC: 'PRICE_DESC', // 높은가격순
});

// 화면에 노출 텍스트
export const SortOptionText = Object.freeze({
    [SortOption.RECOMMENDED]: '추천순',
    [SortOption.LATEST]: '최신순',
    [SortOption.PRICE_ASC]: '낮은가격순',
    [SortOption.PRICE_DESC]: '높은가격순',
});

// 변환 함수
export const getSortOptionText = (option) => SortOptionText[option];
